### PR TITLE
Fix Janitor configuration;

### PR DESCRIPTION
**Pull request by bnjbvr**
Hey @jankeromnes, can you help me please? It's almost working, but I couldn't find out how to start the postgresql daemon from supervisord. Any idea here?

edit: starting from supervisord as I did doesn't work, for some reason (?).

### CODE REVIEW

**Comment by jankeromnes** [comment]
Wow, super cool! Thanks for working on this fix. 😄 

I also don't see anything wrong with your approach. Have you looked at `sudo docker logs <your-test-container>`? This will display the supervisor logs with any errors postresql might throw.

**Comment by bnjbvr** [comment]
Actually, scratch my issue: by cargo culting https://github.com/allisson/docker-postgresql/blob/master/supervisord.conf I've found a way that works.

**Comment by bnjbvr** [comment]
This is ready for review:

- @jankeromnes can you have a look at the Janitor changes, please?
- @Chocobozzz too if you want; there's also a change to the watch:client command so that the front web server is exposed to all interfaces (viz. to the docker sub network). I think it's pretty safe.

Thanks!

File: support/docker/dev/Dockerfile [hunk 1]
```diff
@@ -15,10 +15,22 @@ WORKDIR /home/user/PeerTube
...
+# Install dependencies.
+RUN yarn install
+
 # Configure Janitor for PeerTube.
 ADD janitor.json /home/user/
 RUN sudo chown user:user /home/user/janitor.json
 
 # Configure and build PeerTube.
-RUN yarn install \
- && npm run build
```
Comment (jankeromnes) [inline support/docker/dev/Dockerfile]: Nit: Why do you no longer pre-build PeerTube?

File: support/docker/dev/Dockerfile [hunk 2]
```diff
@@ -15,10 +15,22 @@ WORKDIR /home/user/PeerTube
...
+ADD create_user.sql /tmp/
+RUN sudo service postgresql start && \
+    sudo -u postgres psql --file=/tmp/create_user.sql && \
+    sudo rm /tmp/create_user.sql
```
Comment (jankeromnes) [inline support/docker/dev/Dockerfile]: Nit: Even if you delete files here, they'll remain part of the image history, since they were `ADD`ed above. I suggest leaving this file somewhere useful, or running these steps with inline psql commands as opposed to a file.

Also, if this is really necessary to get PeerTube to work, why isn't it in a `Makefile` or in the `package.json` NPM scripts?

File: support/docker/dev/janitor.json [hunk 3]
```diff
@@ -19,13 +19,17 @@
       "proxy": "https"
     },
     "9000": {
-      "label": "PeerTube",
+      "label": "PeerTube API",
+      "proxy": "https"
+    },
+    "3000": {
```
Comment (jankeromnes) [inline support/docker/dev/janitor.json]: Super-nit: Please sort ports from smallest to largest (i.e. `"3000"` before `"9000"`).

File: support/docker/dev/janitor.json [hunk 4]
```diff
@@ -19,13 +19,17 @@
...
+      "label": "PeerTube front",
```
Comment (jankeromnes) [inline support/docker/dev/janitor.json]: Nit: What about "web app" instead of "front"? I'd find that more explicit/transparent.

File: support/docker/dev/janitor.json [hunk 5]
```diff
@@ -19,13 +19,17 @@
...
       "proxy": "https",
       "preview": true
     }
   },
   "scripts": {
-    "Start PeerTube": "npm start",
+    "Start PeerTube": "npm run dev",
```
Comment (jankeromnes) [inline support/docker/dev/janitor.json]: Interesting, why did you need to switch this? What's the difference between `npm start` and `npm run dev`?

File: support/docker/dev/supervisord.conf [hunk 6]
```diff
@@ -0,0 +1,3 @@
+[program:postgresql]
+user = user
+command = sudo -u postgres /usr/lib/postgresql/9.5/bin/postgres -D /var/lib/postgresql/9.5/main -c config_file=/etc/postgresql/9.5/main/postgresql.conf
```
Comment (jankeromnes) [inline support/docker/dev/supervisord.conf]: Nit: This command is very long. Why not make it shorter by using `user = postgres` like in the supervisor config example you found?

File: support/docker/dev/Dockerfile [hunk 7]
```diff
@@ -15,10 +15,22 @@ WORKDIR /home/user/PeerTube
...
+
+ADD supervisord.conf /tmp/supervisord-extra.conf
+RUN cat /tmp/supervisord-extra.conf | sudo tee -a /etc/supervisord.conf && \
+    sudo rm /tmp/supervisord-extra.conf
+
+EXPOSE 3000
+EXPOSE 9000
```
Comment (jankeromnes) [inline support/docker/dev/Dockerfile]: Nit: You can expose multiple ports in one instruction [like so](https://github.com/JanitorTechnology/dockerfiles/blob/6d3e7835c90fafe76a2e8463b4d5104ac0a39a9d/ubuntu-dev/ubuntu-dev.dockerfile#L224).

File: scripts/watch/client.sh [hunk 8]
```diff
@@ -2,4 +2,4 @@
 
 cd client || exit -1
 
-npm run ng -- server --hmr --host localhost --port 3000
+npm run ng -- server --hmr --host 0.0.0.0 --port 3000
```
Comment (jankeromnes) [inline scripts/watch/client.sh]: This looks a bit scary, because you're exposing something to the world that was limited to `localhost` before. Could you please explain why this is needed? Is this a development server?

File: support/docker/dev/Dockerfile (hunk 1 above)
Comment (Chocobozzz) [reply support/docker/dev/Dockerfile]: `npm run dev` will build the client on the fly

File: support/docker/dev/janitor.json (hunk 5 above)
Comment (Chocobozzz) [reply support/docker/dev/janitor.json]: `npm start` just serves start the built server and serves compiled client files.

`npm run dev` builds the server, run it on port 9000, builds the client on the fly and serves it on port 3000

File: support/docker/dev/Dockerfile [hunk 9]
```diff
@@ -15,10 +15,22 @@ WORKDIR /home/user/PeerTube
 # Configure Cloud9 IDE to use PeerTube's source directory as workspace (-w).
 RUN sudo sed -i "s/-w \/home\/user/-w \/home\/user\/PeerTube/" /etc/supervisord.conf
 
+# Install dependencies.
+RUN yarn install
```
Comment (Chocobozzz) [inline support/docker/dev/Dockerfile]: `RUN yarn install --pure-lockfile`

File: scripts/watch/client.sh (hunk 8 above)
Comment (bnjbvr) [reply scripts/watch/client.sh]: It's not scary, it's only in development mode: note that this just means exposing the dev web server on the Docker network interface, which is needed if we want to expose the service outside of the container. Keeping it this way (the risk seems nonexistent to me).

File: support/docker/dev/Dockerfile (hunk 1 above)
Comment (bnjbvr) [reply support/docker/dev/Dockerfile]: What @Chocobozzz said. Closing.

File: support/docker/dev/Dockerfile (hunk 2 above)
Comment (bnjbvr) [reply support/docker/dev/Dockerfile]: Re: files: well, these are tiny, so I don't seem to care too much ;) I'll just let them in /tmp in this case.

Re: Makefile/package.json docs: it depends a lot on your database install / version; it's more of an external dependency that something mandated by PeerTube, so IMO it makes sense to keep it separate.

File: support/docker/dev/janitor.json (hunk 4 above)
Comment (bnjbvr) [reply support/docker/dev/janitor.json]: If I wanted to be picky, I'd say the API also uses web technologies and is an app, so it's also a web app ;) But yeah, it probably makes more sense overall :)

File: support/docker/dev/supervisord.conf (hunk 6 above)
Comment (bnjbvr) [reply support/docker/dev/supervisord.conf]: I tried it the way you suggest, it didn't work, I don't have much time digging up why it didn't work, so meh 🤷‍. Happy to review a follow-up change to this if you're in the mood of making your PR thereafter.

**Comment by bnjbvr** [comment]
Thanks for the reviews, @Chocobozzz @jankeromnes ! Mind to have another look, please? (this works on my machine; since the basic configuration works, and we still have no guarantees that absolute links will work in Janitor anyways, I'd like to **not** invest too much time on this)

**Comment by Chocobozzz** [comment]
Seems good for me. Waiting @jankeromnes validation.

Even if it does not work yet, we can merge and fix it in another PR :)
